fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -854,4 +859,24 @@ public static CompletableFuture<LucenePartitionInfoProto.LucenePartitionInfo> ge
         }
 
     }
+
+    public static CompletableFuture<LucenePartitionInfoProto.LucenePartitionInfo> getNextNewerPartitionInfo(
+            @Nonnull final FDBRecordContext context,
+            @Nonnull final Tuple groupingKey,
+            @Nullable final Long currentPartitionFromTimestamp,
+            @Nonnull final Subspace indexSubspace) {
+        if (currentPartitionFromTimestamp == null) {
+            return getNewestPartition(groupingKey, context, indexSubspace);
+        } else {
+            final Range range = new TupleRange(
+                    partitionMetadataKeyTuple(groupingKey, currentPartitionFromTimestamp),
+                    partitionMetadataKeyTuple(groupingKey, Long.MAX_VALUE),
+                    EndpointType.TREE_START,
+                    EndpointType.RANGE_EXCLUSIVE)
```

Comment:
This looks wrong, I think you want
```suggestion
                    partitionMetadataKeyTuple(groupingKey, currentPartitionFromTimestamp),
                    partitionMetadataKeyTuple(groupingKey),
                    EndpointType.RANGE_EXCLUSIVE,
                    EndpointType.TREE_END)
```

1. what happens if someone has a timestamp of `Long.MAX_VALUE`?
2. I think the difference between `TREE_START` and `RANGE_EXCLUSIVE` is more semantics than actual behavior difference. But the key is what you want it to return if there's nothing newer. Do you return the current partition, or null.